src/components/panel/panel.js

```diff
@@ -1131,6 +1144,51 @@ MdPanelRef.prototype._configurePosition = function() {
 
 
 /**
+ * Focuses on the panel or the first focus target.
+ * @private
+ */
+MdPanelRef.prototype._focusOnOpen = function() {
+  if (this._config['focusOnOpen']) {
+    // Wait a digest to guarantee md-autofocus has finished adding the class
+    // _md-autofocus, otherwise the focusable element isn't available to focus.
+    var self = this;
+    this._$rootScope.$applyAsync(function() {
```

Comment:
Maybe add as a TODO, but use $$postDigest, which is when angular is done rendering everything and is stable, but doesn't have to wait a full digest where other things can happen.